### PR TITLE
Codegenerate verification

### PR DESCRIPTION
**Pull request by Stagno**
## Technical Description

Adds (to cuda-ico backend) C++ code generation of verification utilities devised in ICON-DSL and their F90 bindings.
Also adds the F90 wrapper `wrap_run_<Stencil_name>()`.
Minor refactors of cuda-ico codegen to share common functions. 

### Testing

Must test this with ICON-DSL

### CODE REVIEW

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,3 @@ option(DAWN_BUILD_TESTING "Enable Dawn tests." ${BUILD_TESTING})
 option(DAWN_BUILD_EXAMPLES "Enable Dawn examples." ${BUILD_TESTING})
 add_subdirectory(dawn)
```
Comment (mroethlin) [inline CMakeLists.txt:13]:
```suggestion
message(STATUS "Configuring GTClang...")
option(GTCLANG_BUILD_TESTING "Enable GTClang tests." ${BUILD_TESTING})
add_subdirectory(gtclang)
```

needs to be added agian, deactivated to be able to build on daint

File: dawn/src/dawn/IIR/Stencil.h [hunk 2]
```diff
@@ -245,6 +245,7 @@ class Stencil : public IIRNode<IIR, Stencil, MultiStage, impl::StdList> {
   /// @{
   void updateFields();
   void updateFields(const Lifetime& lifetime);
+  void updateFieldIntends();
```
Comment (mroethlin) [inline dawn/src/dawn/IIR/Stencil.h:248]: needs to be removed, sorry

File: dawn/test/unit-test/dawn/Optimizer/TestPassCaching.cpp [hunk 3]
```diff
@@ -54,6 +54,8 @@ TEST(TestCaching, test_global_iteration_space_01) {
   PassSetCaches cachingPass;
   cachingPass.run(stencil);
 
+  stencil->computeDerivedInfo();
```
Comment (mroethlin) [inline dawn/test/unit-test/dawn/Optimizer/TestPassCaching.cpp:57]: remove, sorry

File: dawn/test/unit-test/dawn/Optimizer/TestPassSetCaches.cpp [hunk 4]
```diff
@@ -36,6 +36,7 @@ class TestPassSetCaches : public ::testing::Test {
                const std::vector<std::vector<Cache::CacheType>>& cacheTypes,
                const std::vector<std::vector<Cache::IOPolicy>>& ioPolicies) {
     auto instantiation = IIRSerializer::deserialize(filename);
+    instantiation->computeDerivedInfo();
```
Comment (mroethlin) [inline dawn/test/unit-test/dawn/Optimizer/TestPassSetCaches.cpp:39]: remove, sorry

**Review by mroethlin — APPROVED** [review]
some un-needed stuff which slipped through because of unclean diffs, then gtg. sorry for that

**Comment by mroethlin** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Pull request merged**